src/components/Image/Image.tsx

```diff
@@ -1,49 +1,73 @@
+export type ImageProps = React.ImgHTMLAttributes<HTMLImageElement> & {
   isSrcLoading?: boolean;
 } & ({ href: string; title: string } | { href?: never; title?: never });
 
-type Props = {
-  img?: ImageProps;
-  className?: string;
-};
+const Image = React.forwardRef<HTMLImageElement, ImageProps>(
+  ({ className, ...props }, forwardRef) => {
+    const ref = useRef<HTMLImageElement>(null);
+    const [isLoading, setLoading] = useState(!!props.src || props.isSrcLoading);
+    const [isError, setError] = useState(false);
 
-export default function Image({ img, className }: Props) {
-  const ref = useRef<HTMLImageElement>(null);
-  const [isLoading, setLoading] = useState(!!img?.src || img?.isSrcLoading);
+    // https://legacy.reactjs.org/docs/hooks-reference.html#useimperativehandle
+    useImperativeHandle<HTMLImageElement | null, HTMLImageElement | null>(
+      forwardRef,
+      () => ref.current
+    );
 
-  if (!img?.src) {
-    return <ImagePlaceholder className={className} />;
-  }
+    if (!props.src || isError) {
+      return <ImagePlaceholder className={className} />;
+    }
 
-  const shouldLoad = !ref.current?.complete && isLoading;
+    const shouldLoad = !ref.current?.complete && isLoading;
 
-  return (
-    <>
-      {shouldLoad && <ContentLoader className={className} />}
-      {!img.isSrcLoading && (
-        <WithLink
-          className={`${className} ${shouldLoad ? "hidden" : ""}`}
-          href={img.href}
-          title={img.title}
-        >
-          <img
-            ref={ref}
-            src={img.src}
-            className={`${className} object-contain w-full h-full`}
-            alt=""
-            onLoad={() => setLoading(false)}
-          />
-        </WithLink>
-      )}
-    </>
-  );
-}
+    return (
+      <>
+        {shouldLoad && <ContentLoader className={className} />}
+        {!props.isSrcLoading && (
+          /**
+           * Setting the logic to add `hidden` class name is necessary on both
+           * `WithLink` wrapper and on the child `img`.
+           *
+           * Reason:
+           * if no `href` was passed, that means only the image would be returned and since
+           * it is returned without a wrapper, we need to apply `hidden` className manually.
+           * Otherwise (if `href` was passed), we need to apply `hidden` to the link component
+           * wrapping the `img`.
+           */
+          <WithLink
+            className={`${className} ${shouldLoad ? "hidden" : ""}`}
+            href={props.href}
+            title={props.title}
+          >
+            <img
+              ref={ref}
+              src={props.src}
+              className={`object-contain ${
+                shouldLoad ? "hidden" : ""
+              } ${className}`}
+              alt={props.alt || ""}
+              loading={props.loading}
+              onLoad={() => setLoading(false)}
+              onError={(e) =>
+                props.onError ? props.onError(e) : setError(true)
```

Comment:
@SovereignAndrey @ap-justin this logic makes little sense, but it has a reasoning behind it.
The usual way would be to treat `props.onError` as a hook into the error event that is run in addition to any internal error handling, like:
```ts
onError={(e) => {
  if (props.onError) props.onError(e);
  // other internal error logic
  // ...
}}
```
But because I wanted to maintain existing behavior in just one component - `Card`, had to implement `onError` here to be used as a direct replacement for any internal error handling (either use `props.onError` OR use internal error logic).

The existing behavior in question is that `Card` wants to just [append a blue background when an error happens with no placeholder image](https://github.com/AngelProtocolFinance/angelprotocol-web-app/blob/img-to-image/src/pages/Marketplace/Cards/Card.tsx#L39). If we could make it so that the `onError` behavior here is updated to just show the [ImagePlaceholder](https://github.com/AngelProtocolFinance/angelprotocol-web-app/blob/img-to-image/src/components/Image/ImagePlaceholder.tsx) on error, i.e. use the [default `Image.onError` behavior](https://github.com/AngelProtocolFinance/angelprotocol-web-app/blob/img-to-image/src/components/Image/Image.tsx#L27), it would enable us to update `props.onError` to be treated as a hook into the error event (described above).

This would also enable us to remove `onError` in other places where it's used and update them to use some a fallback image. Currently kept their logic as-is as it's less changes to make (see [Leaderboard > Table > Row](https://github.com/AngelProtocolFinance/angelprotocol-web-app/blob/img-to-image/src/pages/Leaderboard/Table/Row/index.tsx#L20) and [Admin > Proposal > DiffTable](https://github.com/AngelProtocolFinance/angelprotocol-web-app/blob/img-to-image/src/pages/Admin/Proposal/Content/Preview/DiffTable.tsx#L54)).